pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -112,15 +112,20 @@ func (c *controller) RegisterEventHandler(typ string, f func(model.Config, model
 			return nil
 		}
 
+		// In 1.0, Pilot has a single function, clearCache, which ignores
+		// the inputs.
+		// In future we may do smarter processing - but first we'll do
+		// major refactoring. No need to recompute everything and generate
+		// multiple events.
+
 		// TODO: This works well for Add and Delete events, but not so for Update:
 		// An updated ingress may also trigger an Add or Delete for one of its constituent sub-rules.
 		switch typ {
 		case model.Gateway.Type:
-			config, _ := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
```

Comment:
nit. If you are going to be adding a gateway for the ingress, you can essentially get rid of the whole Gateway case and have just the model.VirtualService ..